### DISCUSSION TITLE
scene 에서는 스타트 버튼이 보이는데 play 시 버튼이 안보이는 현상

### DISCUSSION THREAD

**Discussion by almondsky** [naverz/zepeto-studio-kor#998] [5. Q&A]
안녕하세요, 
특정위치에 캐릭터가 갔을때 플레이 버튼이 보이고 -> 버튼 클릭시 게임시 작동되도록 구현하려고 하는데,, 위 제목처럼.. 
scene 에서는 스타트 버튼이 보이는데 game play 시 버튼이 안보이는 현상이 생겨 고수님들께 여쭤봅니다.. 
<img width="968" alt="스크린샷 2022-07-13 오전 10 48 40" src="https://user-images.githubusercontent.com/105544091/178631794-0e56ee35-6db0-4781-93da-87ed272abfe3.png">

startgame 버튼은 이렇게 만들어놨구요..  월드를 플레이 하면 감쪽같이 사라져 보이지가 않습니다. 물론 없으니 클릭도 안되구요. 
작성한 typescript 는 이렇습니다,!
<img width="770" alt="스크린샷 2022-07-13 오전 10 50 04" src="https://user-images.githubusercontent.com/105544091/178632312-68a14a4c-7a10-4a35-ae1b-452920a75413.png">

어디서부터 잘못된 것일까요,,,,? 부탁드립니다!

**Comment by YSHFM** [comment]
캔버스 인스펙터와 하이어라키도 혹시 같이 알 수 있을까요?

**Reply by almondsky** [reply]
혹시 몰라 캔버스 | 스타트버튼 캡쳐해서 첨부합니다.
<img width="263" alt="스크린샷 2022-07-13 오전 11 13 53" src="https://user-images.githubusercontent.com/105544091/178636578-bda6cd6b-18f3-4c95-be5a-30b482d280b9.png">
<img width="254" alt="스크린샷 2022-07-13 오전 11 14 31" src="https://user-images.githubusercontent.com/105544091/178636586-0f71cb53-edf8-4c44-b6b1-d2550db5bba0.png">
<img width="268" alt="스크린샷 2022-07-13 오전 11 14 16" src="https://user-images.githubusercontent.com/105544091/178636590-bd813d68-6ed5-41b1-8228-33927a0ffcb1.png">

**Reply by YSHFM** [reply]
만약 스크립트 미사용으로 ui를 항시로 띄워놨을 때도 보이지가 않는다면 
canvas rendermode를 overlay가 아닌 다른 모드를 사용 중이실 때 카메라가 지정되지 않아서 그럴 수 있습니다.
또는 target display나 sort order같은 경우가 잘못되어 있어 렌더링이 안되어 있으실 수 있습니다.

**Reply by almondsky** [reply]
답변 너무 감사합니다,! 제가 꼭 완성하고 싶은 중요한 프로젝트라.. 한가지만 더 여쭤보겠습니다. 말씀하신걸 바탕으로 이것저것 시도해보니 
<img width="276" alt="스크린샷 2022-07-13 오후 12 01 26" src="https://user-images.githubusercontent.com/105544091/178641668-863cc08b-232f-4ead-a6ac-ff9d4a7d94b2.png">
canvas render mode 를 world space 로 바꿨을때 월드 실행시 스타트 버튼 생성이 되었습니다. 앞서 보여드린것과 같이 스크립트는 변동 없구요. 
혹시 몰라 event camera 는 main camera 로 변경해줬습니다. 
스타트 버튼 생성까지는 왔는데,,,, 문제는 스타트 버튼 클릭이 되지 않는데,, 유투브에 나와있는 ui 버튼 클릭안됨 관련 영상들을 보고 따라했는데도 해결이 되지 않아 한번더 여쭤봅니다! 감사합니다!

**Reply by YSHFM** [reply]
zepetoplayers내 카메라를 maincamera로 안바꾸셨다면 먼저 카메라를 변경해주는 것이 첫번 째일 거 같구요.
동일하다면 raycast관련해서 체크해보셔야 될 것 같습니다.

**Reply by YSHFM** [reply]
zepetoplayers내에 카메라를 변경하지 않는다면 카메라가 2개 존재할 수 있습니다.

**Reply by almondsky** [reply]
넵 그렇게 해보겠습니다,! 답변 감사합니다!
